dali/pipeline/data/tensor_vector.cc

```diff
@@ -62,11 +63,112 @@ TensorVector<Backend>::TensorVector(TensorVector<Backend> &&other) noexcept {
   other.tensors_.clear();
 }
 
-
 template <typename Backend>
-size_t TensorVector<Backend>::nbytes() const noexcept {
+void TensorVector<Backend>::UnsafeSetSample(int dst, const TensorVector<Backend> &owner, int src) {
+  // TODO(klecki): more consistency checks, contiguous -> non-contiguous removes shares_data from
+  // samples
+  if (type() == DALI_NO_TYPE && owner.type() != DALI_NO_TYPE) {
+    set_type(owner.type());
+  }
+  if (!order()) {
+    set_order(owner.order());
+  }
+  // Bounds check
+  assert(dst >= 0 && dst < static_cast<int>(curr_tensors_size_));
+  assert(src >= 0 && src < static_cast<int>(owner.curr_tensors_size_));
+  DALI_ENFORCE(type() == owner.type(),
+               make_string("Sample must have the same type as a target batch, current: ", type(),
+                           " new: ", owner.type(), " for ", dst, " <- ", src, "."));
+  DALI_ENFORCE(tensor_shape(dst) == TensorShape<>{0} || sample_dim() == owner.shape().sample_dim(),
+               make_string("Sample must have the same dimensionality as a target batch, current: ",
+                           sample_dim(), " new: ", owner.shape().sample_dim(), " for ", dst, " <- ",
+                           src, "."));
+  DALI_ENFORCE(this->order() == owner.order(), "Sample must have the same order as a target batch");
+  DALI_ENFORCE(
+      GetLayout() == "" || GetLayout() == owner.GetLayout(),
+      make_string("Sample must have the same layout as a target batch current: ", GetLayout(),
+                  " new: ", owner.GetLayout(), " for ", dst, " <- ", src, "."));
+
+  SetContiguous(false);
+  // Setting a new share overwrites the previous one - so we can safely assume that even if
+  // we had a sample sharing into TL, it will be overwritten
+  tensors_[dst]->ShareData(*owner.tensors_[src]);
+  tl_->Reset();
+}
+
+template <typename Backend>
+void TensorVector<Backend>::UnsafeSetSample(int dst, const Tensor<Backend> &owner) {
+  // TODO(klecki): more consistency checks, contiguous -> non-contiguous removes shares_data from
+  // samples
+  if (type() == DALI_NO_TYPE && owner.type() != DALI_NO_TYPE) {
+    set_type(owner.type());
+  }
+  if (!order()) {
+    set_order(owner.order());
+  }
```

Comment:
Likewise - global side-effect on something that looks like it affects just 1 sample.

Reply:
Removed.